dtApi/src/app/login/login.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from '@angular/core'
+import { AuthService } from './services/auth.service'
+import { FormGroup, FormControl } from '@angular/forms'
+import { Router } from '@angular/router'
+import { trigger, style, animate, transition } from '@angular/animations'
+import { loginForm } from './interfaces/interfaces'
+
+@Component({
+    animations: [
+        trigger('inOutAnimation', [
+            transition(':enter', [
+                style({ opacity: 0 }),
+                animate('1s ease-out', style({ opacity: 1 })),
+            ]),
+            transition(':leave', [
+                style({ opacity: 1 }),
+                animate('1s ease-in', style({ opacity: 0 })),
+            ]),
+        ]),
+    ],
+    selector: 'app-login',
+    templateUrl: './login.component.html',
+    styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent implements OnInit {
+    hide = true
+    badRequest = false
+    errorMessage: string
+    loginForm: FormGroup
+    private userName: string
+    private password: string
+
+    constructor(private request: AuthService, private router: Router) {
+        this._createForm()
+    }
+    private _createForm() {
+        this.loginForm = new FormGroup({
```

Comment:
use FormBuilder for reactive forms
https://angular.io/guide/reactive-forms#using-the-formbuilder-service-to-generate-controls